### DISCUSSION TITLE
Updating entity annotations in Spacy NER

### DISCUSSION THREAD

**Discussion by avishwa23** [explosion/spaCy#13243] [Help: Other Questions]
I am having a NER datasets with overlapping/nested entities. I want to perturb the input text of this dataset for adversarial testing, but seems non-trivial to update the ground-truth entities' start and end indices. Is that possible to update the annotations' indices in spacy?

**Comment by Sirivasv** [comment]
What comes to mind, if not already considered, are the Data utilities where you can modify the annotations of data in `.conll` or `.conllu` extensions:
https://spacy.io/usage/training#data-convert

Basically to manipulate nlp objects from it and generate `Doc` objects which have the start_end property:
https://spacy.io/usage/training#custom-code-readers-batchers
